src/components/SignIn/SignInForm.js

```diff
@@ -0,0 +1,146 @@
+      marginLeft: theme.spacing(1),
+      marginRight: theme.spacing(1),
+    },
+  },
+}))(TextField);
+
+const RoundedButton = withStyles((theme) => ({
+  root: {
+    borderRadius: 30,
+    height: theme.typography.fontSize * 3,
+    margin: theme.spacing(3, 0, 2),
+    textTransform: 'none',
+    width: '50%',
+  },
+}))(Button);
+
+const useStyles = makeStyles({
+  container: {
+    display: 'flex',
+    flexDirection: 'column',
+    alignItems: 'center',
+  },
+  buttonWrapper: {
+    display: 'flex',
+    flexDirection: 'column',
+    alignItems: 'center',
+    position: 'relative',
+  },
+  buttonProgress: {
+    position: 'absolute',
+    top: '50%',
+    left: '50%',
+    marginTop: -8,
+    marginLeft: -8,
+  },
+  linksWrapper: {
+    marginTop: 15,
+  },
+  welcome: {
+    marginBottom: 30,
+  },
+});
+
+export default function SignInForm({ setError }) {
+  const classes = useStyles();
+
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [loading, setLoading] = useState(false);
+
+  const handleSubmit = useCallback(() => {
+    setLoading(true);
+    firebase.auth().signInWithEmailAndPassword(email, password)
+      .catch((error) => {
+        setError(error);
+      })
+      .finally(() => {
+        setLoading(false);
+      });
+  }, [email, password, setError]);
+
+  const isInvalid = useMemo(() => password === '' || email === '',
+    [email, password]);
+
+  return (
+    <Container className={classes.container}>
+      <Typography variant="h4" className={classes.welcome}>Welcome back</Typography>
+      <FormControl>
+        <RoundedTextField
+          autoFocus
+          variant="outlined"
+          margin="normal"
+          required
+          fullWidth
+          id="email"
+          label="Email"
+          name="email"
+          autoComplete="email"
+          value={email}
+          onChange={({ target: { value } }) => setEmail(value)}
+          disabled={loading}
+        />
+        <RoundedTextField
+          className={classes.textField}
+          variant="outlined"
+          margin="normal"
+          required
+          fullWidth
+          name="password"
+          label="Password"
+          type="password"
+          id="password"
+          autoComplete="current-password"
+          value={password}
+          onChange={({ target: { value } }) => setPassword(value)}
+          disabled={loading}
+        />
+        <div className={classes.buttonWrapper}>
+          <RoundedButton
+            variant="contained"
+            color="primary"
+            disabled={isInvalid || loading}
+            onClick={handleSubmit}
+          >
+            Sign In
+          </RoundedButton>
+          {loading && <CircularProgress size={24} className={classes.buttonProgress} />}
+        </div>
+        <Grid container justify="space-between" className={classes.linksWrapper}>
+          <Link href={ROUTES.SIGN_UP}>
+            <em>Sign up (devo)</em>
```

Comment:
Is the `(devo)` here intended?

Reply:
Yeah, I wanted to make clear that it's only in development that you can sign up. This is supposed to be "Forgot username?" but that functionality isn't implemented yet. I do agree it's wack, to be honest I think what we could do is just have the link go nowhere for now instead of having this.

Reply:
Removing this link. If users use email to sign in, they don't need 'Forget username?'